### DISCUSSION TITLE
useErrorBoundary and typescript exhaustive check

### DISCUSSION THREAD

**Discussion by AlexKMarshall** [TanStack/query#4124] [Q&A] [answered]
If I'm using `useErrorBoundary: true` on a particularly query, is there any way for typescript to know that my query can't get in an error state?

It looks as though Typescript still thinks that `status === 'error'` is a possibility, even though I believe it should always throw instead. 
I'd like to not have to deal with the error state everywhere I'm using `useQuery` and just delegate it to the boundary.

I put a little codesandbox together that shows the type error if I don't account for the error state. https://codesandbox.io/s/divine-microservice-n9kn3y?file=/src/FetcherComponent.tsx

Is this not designed to work this way? Or is there some other way I can avoid what I think are unnecessary conditionals?

Thanks

**Comment by TkDodo** [comment, marked as answer]
> If I'm using useErrorBoundary: true on a particularly query, is there any way for typescript to know that my query can't get in an error state?

no. we recently did something for `initialData`, so that passing it would take away the `loading` state. However, every time we do this, we need overloads, which complicates the typings. It also only works if it's passed directly to the query. With `initialData`, that's what you'd mostly do, but with `useErrorBoundary`, you often set this globally.

**Reply by AlexKMarshall** [reply]
Thanks very much, that clears things up.
